### ISSUE TITLE
updated stream item fails to render nested stream when in conditionally rendered element

### ISSUE THREAD

**Issue by DaTrader** [phoenixframework/phoenix_live_view#2766]
### Environment

* Elixir version (elixir -v): 1.14.4
* Phoenix version (mix deps): 1.7.6
* Phoenix LiveView version (mix deps): 0.19.5+ commit: 8c2b2be138ee8476c163a5a8748becbc4db7d491
* Operating system: Linux Mint
* Browsers you attempted to reproduce this bug on (the more the merrier):  FF, Chrome
* Does the problem persist after removing "assets/node_modules" and trying again? Yes/no: Yes

### Actual behavior

Trying to circumvent another bug (see #2657), I conditionally render the element with Alpine code (Alpine being irrelevant for this bug here) so I first append the item only to move it to the beginning of the stream in the following update cycle. It works in terms of circumventing the Alpine/LiveView integration issue, but there's this new bug with LiveView not rendering the stream nested within a conditionally rendered element. My guess is LiveView sees the stream itself has not changed and fails to realize it was first within the element that was not rendered because it's condition was not met, while in the second attempt the condition is met and the element is rendered but it skips rendering the stream.

The link to the specific repo branch:
https://github.com/DaTrader/lv_bugs_0_19_5/tree/bug_nested_stream_existing_element

Steps:
1. http://localhost:4000/live
2. Click the "Prepend a random item" below the list.

Although the nested stream itself is not rendered at first, its state is probably ok for as soon as the "Reset colors" button is clicked the nested stream will appear as well as start showing up one color at a time if the "Start carousel" button is clicked first.

### Expected behavior

Obviously, the nested stream should be rendered when its conditionally rendered parent element is.

**Comment by josevalim** [comment]
I am not sure if this can be fixed. You are conditionally rendering a stream and LiveView does not track if the stream was actually rendered or not. Even if we somehow could track assigns or track the diff to see which streams are part of the diff, this may be too expensive.

I will investigate the feasibility of fixing this but for now you should track when prepended? changes and build the stream when prepended swaps value.

**Comment by josevalim** [comment]
I have improved the docs to make it clear it is after every render/1 call. We could clear streams lazily but we chose not to for two reasons:

1. Other features, such as temporary assigns, are also clearerd every render/1 and, opposite to streams, they could not be tackled

2. If we accumulate streams in memory, there is a chance someone forgets to render them and the stream now becomes a memory leak, which would be very surprising, given their goal is to flush data as soon as possible

Therefore, you must assign the streams when they are ready to render.
